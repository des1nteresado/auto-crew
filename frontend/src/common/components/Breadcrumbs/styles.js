const styles = (theme) => ({
  breadcrumbs: {
    fontSize: 12,
  },
});

export default styles;
