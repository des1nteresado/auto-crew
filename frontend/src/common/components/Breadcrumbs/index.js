import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumbs as MaterialBreadcrumbs, Link, withStyles } from '@material-ui/core';

import ROUTES from 'modules/Navigation/routes';

import styles from './styles.js';

const Breadcrumbs = ({ children, classes }) => {
  return (
    <MaterialBreadcrumbs className={classes.breadcrumbs} separator="&#8226;">
      <Link color="textPrimary" href={ROUTES.HOME}>
        Home
      </Link>
      {children}
    </MaterialBreadcrumbs>
  );
};

Breadcrumbs.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  children: PropTypes.node.isRequired,
};

export default React.memo(withStyles(styles)(Breadcrumbs));
