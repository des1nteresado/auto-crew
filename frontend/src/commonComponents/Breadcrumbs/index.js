import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumbs as MaterialBreadcrumbs, Link, withStyles } from '@material-ui/core';
import { useSelector } from 'react-redux';
import { startCase } from 'lodash';

import ROUTES from 'routes';

import styles from './styles.js';

const Breadcrumbs = ({ children, classes }) => {
  const { filterByPeopleCategory } = useSelector((state) => state.global);

  return (
    <MaterialBreadcrumbs className={classes.breadcrumbs} separator="&#8226;">
      <Link color="textPrimary" href={ROUTES.HOME}>
        {startCase(filterByPeopleCategory)}
      </Link>
      {children}
    </MaterialBreadcrumbs>
  );
};
Breadcrumbs.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  children: PropTypes.node.isRequired,
};

export default React.memo(withStyles(styles)(Breadcrumbs));
