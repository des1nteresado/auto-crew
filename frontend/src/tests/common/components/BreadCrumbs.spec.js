import React from 'react';

import Breadcrumbs from 'common/components/Breadcrumbs';

const props = {
  children: <div>Test</div>,
};

const setUp = (props) => shallow(<Breadcrumbs {...props} />);

describe('Breadcrumbs', () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });

  it('renders Breadcrumbs component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('rendrers Breadcrumbs children prop correclty', () => {
    const div = component.find('div');

    expect(div).toHaveLength(1);
    expect(div.text()).toEqual('Test');
  });
});
