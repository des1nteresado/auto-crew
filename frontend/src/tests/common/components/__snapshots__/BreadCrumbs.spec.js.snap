// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders Breadcrumbs component correctly 1`] = `
<Breadcrumbs
  classes={
    Object {
      "breadcrumbs": "Breadcrumbs-breadcrumbs-1",
    }
  }
>
  <div>
    Test
  </div>
</Breadcrumbs>
`;
